Coding/fw/lib/Settings/WLAN.cpp

```diff
@@ -0,0 +1,211 @@
+/**
+*   Determines the state of the WiFiModeSelect Button to enter AP Mode
+*   @return bool AP Mode. If true, will request the AP mode. If false, requests STA Mode
+*/
+static void readWiFiMode();
+
+/**
+* Connects to the WiFi AP when requested
+* @return true when succesfully connected. False otherwise.
+*/
+static bool connectWiFi();
+
+/* VARIABLES **************************************************************************************/
+IPAddress m_serverIP; /**< Stores the IP Address of the ESP32 */
+bool m_APMode = true;
+
+/* PUBLIC METHODES ********************************************************************************/
+/**************************************************************************************************/
+const String &wlan::getSTA_SSID()
+{
+    return STA_SSID;
+}
+/**************************************************************************************************/
+const String &wlan::getSTA_PASS()
+{
+    return STA_PASSWORD;
+}
+
+/**************************************************************************************************/
+const String &wlan::getAP_SSID()
+{
+    return AP_SSID;
+}
+
+/**************************************************************************************************/
+const String &wlan::getAP_PASS()
+{
+    return AP_PASSWORD;
+}
+
+/**************************************************************************************************/
+const bool &wlan::getAP_MODE()
+{
+    return m_APMode;
+}
+
+/**************************************************************************************************/
+bool wlan::begin()
+{
+    bool success = true;
+    Board::apLED.write(HIGH);
+    Board::staLED.write(HIGH);
+
+    Settings::get(DIRECTORY, "STA_SSID", STA_SSID, "");
+    Settings::get(DIRECTORY, "STA_Password", STA_PASSWORD, "");
+
+    readWiFiMode();
+    if (m_APMode)
+    {
+        Board::staLED.write(LOW);
+        if (!WiFi.softAP(AP_SSID.c_str(), AP_PASSWORD.c_str()))
+        {
+            success = false;
+        }
+
+        m_serverIP = WiFi.softAPIP();
+    }
+    else
+    {
+        Board::apLED.write(LOW);
+        if (!WiFi.mode(WIFI_STA))
+        {
+            success = false;
+        }
+        else if (WiFi.begin(STA_SSID.c_str(), STA_PASSWORD.c_str()) == WL_CONNECT_FAILED)
+        {
+            success = false;
+        }
+        else if (!connectWiFi())
+        {
+            success = false;
+        }
+
+        m_serverIP = WiFi.localIP();
+    }
+    return success;
+}
+
+/**************************************************************************************************/
+/**
+*   Calls next request on DNS Server
+*   return true if a restart is requested
+*/
+bool wlan::checkConnection()
+{
+    bool success = true;
+    if (WiFi.getMode() == WIFI_STA && WiFi.status() != WL_CONNECTED)
+    {
+        if (!connectWiFi())
+        {
+            success = false;
+        }
+    }
+    return success;
+}
+
+IPAddress wlan::getIPAddress()
```

Comment:
as a general rule, always return objects as const& unleass you really want a copy to be returned. Which is rare.